apollo-federation/src/sources/connect/validation/schema.rs

```diff
@@ -327,14 +345,145 @@ fn advanced_validations(schema: &SchemaInfo, subgraph_name: &str) -> Vec<Message
     entity_checker.check_for_missing_entity_connectors(schema)
 }
 
+fn compute_input_trie(connector: &Connector) -> SelectionTrie {
+    let mut merged = SelectionTrie::new();
+    for param in connector.variable_references() {
```

Comment:
this can probably be refactor in an iterator way:

```rust
connector.variable_references().iter().ma(|param| param.selection).collect::<SelectionTrie<_>>()
```

Reply:
But… why? It's more code. Can't Rust generate efficient code for the `for` loop?